index.bs

```diff
@@ -52,10 +70,192 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">open(|name|, |options|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+    
+1. Let |shelf| be the result of running [=obtain a local storage shelf=] given |environment|.
+
+1. If |shelf| is failure, then return [=a promise rejected with=] a {{TypeError}}.
+
+1. If the result of [=validate a bucket name=] with |name| is failure,  then return [=a promise rejected with=] a {{TypeError}}.
+
+1. Let |r| be the result of running [=open a bucket=] with |shelf|, |name|, and |options|.
+
+1. If |r| is failure, then return [=a promise rejected with=] a {{TypeError}}.
+
+1. Return [=a promise resolved with=] with |r|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>open a bucket</dfn> for a |shelf| given a bucket |name| and optional |options|, run the following steps:
+
+1. Let |expires| be undefined.
+    
+1. If |options|["{{StorageBucketOptions/expires}}"] exists, then:
+    
+    1. Let |expires| be the result of [=parse a duration string=] from |options|["{{StorageBucketOptions/expires}}"].
```

Comment:
Do we need this parsing step?  Its now defined as a DOMHighResTimeStamp and not a string.